dao-contracts/src/voting/governance_voting/voting.rs

```diff
@@ -0,0 +1,361 @@
+use casper_dao_utils::Address;
+use casper_types::{
+    bytesrepr::{FromBytes, ToBytes},
+    CLType, CLTyped, RuntimeArgs, U256,
+};
+
+use crate::voting::vote::VotingId;
+
+pub enum VotingResult {
+    InFavor,
+    Against,
+    QuorumNotReached,
+}
+
+pub enum VotingType {
+    Informal,
+    Formal,
+}
+
+pub struct VotingConfiguration {
+    pub formal_voting_quorum: U256,
+    pub formal_voting_time: u64,
+    pub informal_voting_quorum: U256,
+    pub informal_voting_time: u64,
+    pub minimum_governance_reputation: U256,
+    pub contract_to_call: Option<Address>,
+    pub entry_point: String,
+    pub runtime_args: RuntimeArgs,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct Voting {
+    voting_id: VotingId,
+    completed: bool,
+    stake_in_favor: U256,
+    stake_against: U256,
+    start_time: u64,
+    informal_voting_id: VotingId,
+    formal_voting_id: Option<VotingId>,
+    formal_voting_quorum: U256,
+    formal_voting_time: u64,
+    informal_voting_quorum: U256,
+    informal_voting_time: u64,
+    minimum_governance_reputation: U256,
+    contract_to_call: Option<Address>,
+    entry_point: String,
+    runtime_args: RuntimeArgs,
+}
+
+impl Voting {
+    pub fn new(
+        voting_id: U256,
+        start_time: u64,
+        voting_configuration: VotingConfiguration,
+    ) -> Self {
+        Voting {
+            voting_id,
+            completed: false,
+            stake_in_favor: U256::zero(),
+            stake_against: U256::zero(),
+            start_time,
+            informal_voting_id: voting_id,
+            formal_voting_id: None,
+            formal_voting_quorum: voting_configuration.formal_voting_quorum,
+            formal_voting_time: voting_configuration.formal_voting_time,
+            informal_voting_quorum: voting_configuration.informal_voting_quorum,
+            informal_voting_time: voting_configuration.informal_voting_time,
+            minimum_governance_reputation: voting_configuration.minimum_governance_reputation,
+            contract_to_call: voting_configuration.contract_to_call,
+            entry_point: voting_configuration.entry_point,
+            runtime_args: voting_configuration.runtime_args,
+        }
+    }
+
+    pub fn get_voting_type(&self) -> VotingType {
+        if self.voting_id == self.informal_voting_id {
+            VotingType::Informal
+        } else {
+            VotingType::Formal
+        }
+    }
+
+    pub fn convert_to_formal(&self, new_voting_id: U256, start_time: u64) -> Self {
```

Comment:
It's misleading to me. It does not covert the current object, it clones and makes some changes, but the original struct remains the same. I would rather name it `clone_as_formal` or something